galley/pkg/config/processor/transforms/transforms.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package transforms contains basic processing building blocks that can be incorporated into bigger/self-contained
+// processing pipelines.
+package transforms
+
+import (
+	"istio.io/istio/galley/pkg/config/collection"
+	"istio.io/istio/galley/pkg/config/event"
+	"istio.io/istio/galley/pkg/config/processing"
+)
+
+//TODO: doc comments everywhere
+type Info struct {
+	inputs   collection.Names //TODO: Combine these into a struct?
```

Comment:
Combining inputs/outputs into its struct allows you to use the same struct in the schema package.
For example, you can have:
```golang
package schema

type TransformInfo struct {
    Inputs   collection.Names
   Outputs collection.Names
}
```

and
```golang
package transforms

// Replaces the current Info struct
type Provider struct {
   Info schema.TransformInfo
   Create func(_) event.Transformer
}
```

Reply:
I'm inclined to pass on this for now: this change is big enough already.